srcs/runners_manager/vm_creation/gcloud/GcloudManager.py

```diff
@@ -98,6 +98,7 @@ def configure_instance(
             project=runner.vm_type.config["project"],
             family=runner.vm_type.config["family"],
         )
+        self.zone = runner.vm_type.config.get("zone", self.zone)
```

Comment:
I don't know if we should overwrite the self here. There might be some unwanted impact. Could you have a look into that?

We also need to take the whole code into account, per example I see on the `update_vm_metadata` function we use `self.zone` to get the instances, chances are that `self.zone` and the actual zone of the instance are not the same, and the instances.get method will return that the instance does not exist (because it doesn't in that zone).

Same story for get_all_vms method.

If zone is a optional argument in those cases where we are retrieving an instance by name or listing them, then it will help us solve the issue.